trio/__init__.py

```diff
@@ -12,87 +12,104 @@
 #
 # This file pulls together the friendly public API, by re-exporting the more
 # innocuous bits of the _core API + the higher-level tools from trio/*.py.
+#
+# Uses `from x import y as y` for compatibility with `pyright --verifytypes` (#2625)
 
-from ._version import __version__
+# pyright explicitly does not care about `__version__`
+# see https://github.com/microsoft/pyright/blob/main/docs/typed-libraries.md#type-completeness
+from ._version import __version__ as __version__
```

Comment:
The comment yet having `import x as x` is kinda confusing me. Did you just forget to revert the `import x as x` change here?